### PR TITLE
✨ Add engame behaviour and metal bricks durability 

### PR DESCRIPTION
**Pull request by maelehrhard**

### CODE REVIEW

File: arkanoid.c [hunk 1]
```diff
@@ -97,7 +78,7 @@ int main(int argc, char **argv) {
             move_VAUS(10, 0);
         }
         if (keys[SDL_SCANCODE_SPACE]) {
-            if (lives > 0) {
+            if (lives > 0 && !is_end_game()) {
```
Comment (axelmunch) [inline arkanoid.c:81]: Apply the same test on player 2, line 108

File: graphics.c [hunk 2]
```diff
@@ -153,49 +169,58 @@ void draw_lives(SDL_Surface *win_surf, int lives) {
...
-        dead_text_width =
-            draw_text(win_surf, "Press SPACE to restart",
-                      win_surf->w / 2 - dead_text_width / 2, win_surf->h / 2);
+        draw_score(win_surf);
+
+        if (DEBUG_MODE) {
+            draw_text(win_surf, "FPS", 10, win_surf->h - 74);
+            draw_integer(win_surf, (int) get_current_fps(), 10,
+                         win_surf->h - 42);
+        }
```
Comment (axelmunch) [inline graphics.c]: Do not condition FPS displaying. Move this at the end after the is_end_game test.

File: entities/capsule.c [hunk 3]
```diff
@@ -52,6 +52,10 @@ void apply_divide_capsule() {
         play_chunk(DIVIDE);
     }
 }
+void apply_break_capsule(SDL_Surface *win_surf) {
+    load_next_level(win_surf);
+    play_chunk(BREAK);
```
Comment (axelmunch) [inline entities/capsule.c]: Can you make it so the sound also plays when we finish a level without this capsule (move play_chunk inside of load_next_level function)

File: arkanoid.c [hunk 4]
```diff
@@ -166,21 +146,13 @@ int main(int argc, char **argv) {
...
                 cheat_key_press = true;
                 apply_addition_capsule();
             }
-        } else if (keys[SDL_SCANCODE_8]) {
-            if (!cheat_key_press && DEBUG_MODE) {
-                cheat_key_press = true;
-                reset_score();
-                restart_level_1();
-                load_next();
-            }
```
Comment (axelmunch) [inline arkanoid.c]: Why did you remove this?

File: graphics.c [hunk 5]
```diff
@@ -43,6 +43,22 @@ void draw_background(SDL_Surface *win_surf) {
...
+    int mock;
+    int black_background_width, black_background_height;
+    get_texture_dimensions(BlackBackground, &mock, &mock,
+                           &black_background_width, &black_background_height);
+    for (int i = SCREEN_WIDTH - black_background_width;
+         i > -black_background_width; i -= 32) {
+        for (int j = 0; j < win_surf->h; j += black_background_height) {
+            draw_texture(win_surf, BlackBackground, i, j, false);
+        }
+    }
```
Comment (axelmunch) [inline graphics.c]: Using surface fill here may be more efficient:
```suggestion
    SDL_FillRect(win_surf, NULL, SDL_MapRGB(win_surf->format, 0, 0, 0));

```

File: graphics.c [hunk 6]
```diff
@@ -43,6 +43,22 @@ void draw_background(SDL_Surface *win_surf) {
...
+    dead_text_width =
+        draw_text(win_surf, "VICTORY ! Press SPACE to restart",
+                  win_surf->w / 2 - dead_text_width / 2, win_surf->h / 2);
+}
```
Comment (axelmunch) [inline graphics.c:53]: Please add `draw_score(win_surf);` on the victory screen.

File: arkanoid.c (hunk 4 above)
Comment (maelehrhard) [reply arkanoid.c]: because we only have 7 capsules. This was a duplicate from numpad 6

File: arkanoid.c (hunk 1 above)
Comment (maelehrhard) [reply arkanoid.c:81]: done

File: graphics.c (hunk 2 above)
Comment (maelehrhard) [reply graphics.c]: ok

File: entities/capsule.c (hunk 3 above)
Comment (maelehrhard) [reply entities/capsule.c]: yes. it now plays on every level change except at the beginning and the end

File: graphics.c (hunk 6 above)
Comment (maelehrhard) [reply graphics.c:53]: yes

File: arkanoid.c (hunk 4 above)
Comment (axelmunch) [reply arkanoid.c]: This key is not linked to a capsule and is to reset the game at level 1.

**Review by axelmunch — APPROVED** [review]
Nice
